lsp/base_service_client/base_service_client.go

```diff
@@ -684,6 +717,73 @@ func (sc *LSPServiceClientBase) processSymbolCacheUpdate(fileURI uri.URI) {
 	sc.populateDocumentSymbolCache(sc.Ctx, []uri.URI{fileURI})
 }
 
+// reportProgress reports progress to the progress reporter if one is configured.
+// It throttles updates to avoid excessive reporting (max once per 500ms).
+// It also sends events to the progress stream channel if streaming is active.
+func (sc *LSPServiceClientBase) reportProgress() {
```

Comment:
It looks like golang service clients are going to need to implement all of this similar logic, would it be possible to create a "StreamableProgressReporter" or something in the progress package and encapsulate all the logic and then the service clients can use the progress reporter the same way as in the analyzer-lsp and it handles the batching/timing